ports/raspberrypi/common-hal/audiopwmio/PWMAudioOut.c

```diff
@@ -126,17 +126,20 @@ void common_hal_audiopwmio_pwmaudioout_deinit(audiopwmio_pwmaudioout_obj_t *self
 }
 
 void common_hal_audiopwmio_pwmaudioout_play(audiopwmio_pwmaudioout_obj_t *self, mp_obj_t sample, bool loop) {
-    if (common_hal_audiopwmio_pwmaudioout_get_playing(self)) {
-        common_hal_audiopwmio_pwmaudioout_stop(self);
+
+    if (self->dma.channel[0] < NUM_DMA_CHANNELS) {
+        if (common_hal_audiopwmio_pwmaudioout_get_playing(self)) {
```

Comment:
Note that ...get_playing checks for the dma channel number:
```c
bool audio_dma_get_playing(audio_dma_t *dma) {
    if (dma->channel[0] == NUM_DMA_CHANNELS) {
        return false;
    }
```
so this added check may be unnecessary.  If it is necessary, please add a source comment why and check if PWMAudioOut needs the added check too.

Reply:
The issue is that in  common_hal_audiopwmio_pwmaudioout_get_playing() we are accessing self->pacing_timer before it has been set. So, the second PWMAudioOut will reset the timer (and so stop the first one). I could either initialize self->pacing timer in audio_dma_init() or leave the current code as is. Which do you think would be cleaner?